### PR TITLE
traffic-gen, testpmd: Fix number of streams created

### PR DESCRIPTION
**Pull request by RamLavi**
Currently the traffic generator is creating too many streams. Fixing the script template to create a stream per every traffic generating CPU.

Fixed BZ#2170437

### CODE REVIEW

File: traffic-gen/scripts/set_tests_files.sh [hunk 1]
```diff
@@ -30,7 +30,7 @@ print_params() {
 	echo SRC_EAST_MAC_ADDRESS="${SRC_EAST_MAC_ADDRESS}"
 	echo DST_WEST_MAC_ADDRESS="${DST_WEST_MAC_ADDRESS}"
 	echo DST_EAST_MAC_ADDRESS="${DST_EAST_MAC_ADDRESS}"
-	echo NUM_OF_CPUS="${NUM_OF_CPUS}"
+	echo NUM_OF_TRAFFIC_CPUS="${NUM_OF_TRAFFIC_CPUS}"
```
Comment (orelmisan) [inline traffic-gen/scripts/set_tests_files.sh:33]: How about `ISOLATED_CPU_COUNT`?

Comment (RamLavi) [reply traffic-gen/scripts/set_tests_files.sh:33]: it will be too ambiguous.

**Comment by RamLavi** [comment]
```
zeus09:kubevirt-dpdk-checkup $ TEST_NAMESPACE=ralavi-ns make TEST_IMAGE=quay.io/ramlavi/kubevirt-dpdk-checkup:latest TRAFFIC_GEN_IMAGE_URL=quay.io/ramlavi/kubevirt-dpdk-checkup-traffic-gen:latest test/e2e 
docker run --rm \
           --volume /root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup:/root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup:Z \
           --volume /root/.kube:/root/.kube:Z \
           --workdir /root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup \
           -e KUBECONFIG=/root/.kube/config \
           -e TEST_IMAGE=quay.io/ramlavi/kubevirt-dpdk-checkup:latest \
           -e TEST_NAMESPACE=ralavi-ns \
           -e NETWORK_ATTACHMENT_DEFINITION_NAME= \
           -e RUNTIME_CLASS_NAME= \
           -e TRAFFIC_GEN_IMAGE_URL=quay.io/ramlavi/kubevirt-dpdk-checkup-traffic-gen:latest \
           -e VM_CONTAINER_DISK_IMAGE_URL= \
           docker.io/library/golang:1.19 go test ./tests/... -test.v -test.timeout=1h -ginkgo.v -ginkgo.timeout=1h
=== RUN   TestKubevirtDpdkCheckup
Running Suite: KubevirtDpdkCheckup Suite - /root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup/tests
=======================================================================================================
Random Seed: 1676536118

Will run 1 of 1 specs
------------------------------
[BeforeSuite] 
/root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup/tests/test_suite_test.go:63
[BeforeSuite] PASSED [0.004 seconds]
------------------------------
Execute the checkup Job should complete successfully
/root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup/tests/checkup_test.go:79
  [FAILED] in [It] - /root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup/tests/checkup_test.go:89 @ 02/16/23 08:31:24.019
• [FAILED] [165.775 seconds]
Execute the checkup Job [It] should complete successfully
/root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup/tests/checkup_test.go:79

  [FAILED] checkup failed: {
  	"spec.param.networkAttachmentDefinitionName": "intel-dpdk-network-1",
  	"spec.param.testDuration": "1m",
  	"spec.param.trafficGeneratorImage": "quay.io/ramlavi/kubevirt-dpdk-checkup-traffic-gen:latest",
  	"spec.param.trafficGeneratorRuntimeClassName": "performance-performance-zeus10",
  	"spec.param.verbose": "true",
  	"spec.param.vmContainerDiskImage": "",
  	"spec.timeout": "10m",
  	"status.completionTimestamp": "2023-02-16T08:31:16Z",
  	"status.failureReason": "detected 840000006 Output Error Packets on the traffic generator's side",
  	"status.result.DPDKRxPacketDrops": "0",
  	"status.result.DPDKTxPacketDrops": "0",
  	"status.result.DPDKVMNode": "zeus10.lab.eng.tlv2.redhat.com",
  	"status.result.trafficGeneratorMaxDropRate": "0.000000",
  	"status.result.trafficGeneratorNode": "zeus10.lab.eng.tlv2.redhat.com",
  	"status.result.trafficGeneratorOutputErrorPackets": "840000006",
  	"status.startTimestamp": "2023-02-16T08:28:49Z",
  	"status.succeeded": "false"
  }
  In [It] at: /root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup/tests/checkup_test.go:89 @ 02/16/23 08:31:24.019
------------------------------

Summarizing 1 Failure:
  [FAIL] Execute the checkup Job [It] should complete successfully
  /root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup/tests/checkup_test.go:89

Ran 1 of 1 Specs in 165.779 seconds
FAIL! -- 0 Passed | 1 Failed | 0 Pending | 0 Skipped
--- FAIL: TestKubevirtDpdkCheckup (165.78s)
FAIL
FAIL	github.com/kiagnose/kubevirt-dpdk-checkup/tests	165.811s
FAIL
make: *** [Makefile:59: test/e2e] Error 1
```

File: traffic-gen/templates/testpmd.py.in [hunk 2]
```diff
@@ -31,9 +31,9 @@ class STLS1(object):
 
 
     def get_streams (self, direction = 0, **kwargs):
-        # create multiple streams, one stream per core...
+        # create multiple streams, one stream per core generating traffic...
         s = []
-        for i in range($NUM_OF_CPUS-1):
+        for i in range($NUM_OF_TRAFFIC_CPUS):
```
Comment (orelmisan) [inline traffic-gen/templates/testpmd.py.in:36]: Isn't the number of created stream now larger instead of smaller?

Comment (RamLavi) [reply traffic-gen/templates/testpmd.py.in:36]: no, before it was $NUM_OF_CPUS-1 where [NUM_OF_CPUS](https://github.com/kiagnose/kubevirt-dpdk-checkup/blob/main/pkg/internal/checkup/checkup.go#L400) is 8 so there were 7 streams.
Now there are [NUM_OF_TRAFFIC_CPUS](https://github.com/kiagnose/kubevirt-dpdk-checkup/blob/main/pkg/internal/checkup/checkup.go#L399) which is 6 so there we'll have 6 streams.

Comment (orelmisan) [reply traffic-gen/templates/testpmd.py.in:36]: Where is it changed to 6?

Comment (RamLavi) [reply traffic-gen/templates/testpmd.py.in:36]: `NUM_OF_TRAFFIC_CPUS` is an env var we are setting on checkup package ([here](https://github.com/kiagnose/kubevirt-dpdk-checkup/blob/main/pkg/internal/checkup/checkup.go#L399)). We are already using for starting the trex on pod startup (see [here](https://github.com/kiagnose/kubevirt-dpdk-checkup/blob/main/traffic-gen/scripts/run_traffic_gen_daemon.sh#L32)). 
The env var's value is the number of CPUs (=8) minus the number of non traffic nodes (=2).

**Review by orelmisan — APPROVED** [review]

**Pull request merged**